scout/server/blueprints/variants/views.py

```diff
@@ -62,15 +59,32 @@ def variants(institute_id, case_name):
             'gene_panels': clinical_filter_panels
              })
 
+    user_obj = store.user(current_user.email)
     if(request.method == "POST"):
+        # If special filter buttons were selected:
         if bool(request.form.get('clinical_filter')):
             form = FiltersForm(clinical_filter)
             form.csrf_token = request.args.get('csrf_token')
+        elif bool(request.form.get('save_filter')):
```

Comment:
What about a completely different endpoint which is dedicated to save filters? So you can use it from 3 types of variants without repeating the same code in each of them. It could save the filter and return to the request refererrer (variants page)

Reply:
Hm, I did start that at some point(😅), but ditched it since it gets a little involved with passing the info between different forms. It would be possible I guess, but I see it as rather complex whereas this is straight to the point. It does make the veeeery long variants view longer: I think I'll try breaking it out into the controller instead.

Reply:
That's a very good idea. It will make the variants code in views much more readable!